kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/WindowDoc.java

```diff
@@ -47,10 +50,12 @@ public WindowDoc() {
   public WindowDoc(
       @BsonProperty(ID) BasicDBObject id,
       @BsonProperty(VALUE) byte[] value,
+      @BsonProperty(VALUES) List<byte[]> values,
```

Comment:
I was less concerned with the overhead of the list and more about maintaining a clean separation so we can make sure the correct column has data for the given store type

For one thing, if we use a list for both, then for non-duplicates we presumably want to overwrite the entire list on each update or pop the current value then push the new one, which both feel kind of silly/inefficient (just my intuition, maybe it's not that bad 🤷‍♀️ ). But we definitely don't want an ever-increasing list of values for non-duplicate stores, since it can get quite long with many frequent updates in an aggregation window. So we'd still have to branch on `retainDuplicates` on the write path. 

For another thing, and maybe this is a little paranoid, but what if someone has a store with many duplicates and at some point they accidentally run their app with `retainDuplicates` set to false? That could easily wipe out their existing data if it starts overwriting these lists rather than pushing new items. Keeping the fields separate prevents this from happening since each store type gets its own field